core/src/main/java/hudson/security/HudsonAuthenticationEntryPoint.java

```diff
@@ -80,7 +81,7 @@ public void commence(ServletRequest request, ServletResponse response, Authentic
             loginForm = MessageFormat.format(loginForm, URLEncoder.encode(uriFrom,"UTF-8"));
             req.setAttribute("loginForm", loginForm);
 
-            rsp.setStatus(SC_FORBIDDEN);
+            rsp.setStatus(HttpServletResponse.SC_TEMPORARY_REDIRECT);
```

Comment:
regardless of any other comments - where does the `Location` header get set that the client should use for the redirect?  I could not see it in the code - and if performing a redirect like this then you shouldn't (need to) include the HTML that does the redirect below.

Reply:
Also I think this should be using `302` - as we do not want the browser to POST if this was from a POST request at the new URL...